### PR TITLE
Fix JWT token encoding issue in test.

### PR DESCRIPTION
**Pull request by hlinnaka**
On my laptop, the server was receiving the token as a string with extra
b'...' escaping, e.g as "b'eyJ0....0ifQA'" instead of just "eyJ0....0ifQA".
That was causing the test to fail.

I'm using Python 3.9, while the CI is using Python 3.8. I suspect that's
why. My version of pyjwt might be different too.

See also https://github.com/jpadilla/pyjwt/issues/391.

### CODE REVIEW

**Comment by LizardWizzard** [comment]
That's strange, Pipfile.lock should ensure that we have the same version of all packages in all environments. I use python 3.9 locally too. Is it reasonable to dive deeper into this?

**Comment by hlinnaka** [comment]
> That's strange, Pipfile.lock should ensure that we have the same version of all packages in all environments. I use python 3.9 locally too. Is it reasonable to dive deeper into this?

I use the version of python and pyjwt that I get with `apt install ...`, not pipenv stuff. Not a very supported or recommended configuration I guess, so no need to dig deeper.

File: test_runner/batch_others/test_auth.py [hunk 1]
```diff
@@ -22,6 +22,25 @@ class AuthKeys:
     pub: bytes
     priv: bytes
 
+    def generate_management_token(self):
+        token = jwt.encode({"scope": "pageserverapi"}, self.priv, algorithm="RS256")
+
+        # jwt.encode can return 'bytes' or 'str', depending on Python version or type
+        # hinting or something (not sure what). If it returned 'bytes', convert it to 'str'
+        # explicitly.
+        if isinstance(token, bytes):
```
Comment (LizardWizzard) [inline test_runner/batch_others/test_auth.py:31]: nit: maybe split encoding and bytes/string check into its own function, and in this shortcut functions pass body of the token to this generic encode function

**Review by LizardWizzard — APPROVED** [review]
Typically apt have older versions, may be that's the case here. So lets just merge this

**Comment by lubennikovaav** [comment]
On my laptop I caught a similar error in `test_pageserver_auth`:

```Aug 12 20:38:14.311 ERRO error: auth failed: Base64 error: Invalid byte 39, offset 1., location: pageserver::page_service, pageserver/src/page_service.rs:159```

Is it the same problem or something new?

**Comment by hlinnaka** [comment]
> On my laptop I caught a similar error in `test_pageserver_auth`:
> 
> `Aug 12 20:38:14.311 ERRO error: auth failed: Base64 error: Invalid byte 39, offset 1., location: pageserver::page_service, pageserver/src/page_service.rs:159`
> 
> Is it the same problem or something new?

Same problem!

**Pull request merged**